pkg/gpioControl/gpioControlChardev.go

```diff
@@ -0,0 +1,170 @@
+package gpioControl
+
+import (
+	"3mdeb/RteCtrl/pkg/config"
+	"fmt"
+
+	"github.com/warthog618/go-gpiocdev"
+)
+
+type GpioChardev struct {
+	sysGpioPath string
+	gpios       map[int]pin
+}
+
+func (ctrl *GpioChardev) GetPinNumByID(id int) int {
+	return int(ctrl.gpios[id].sysNum)
+}
+
+func RequestLine(sysGpio int, options ...gpiocdev.LineReqOption) (*gpiocdev.Line, error) {
+  var line *gpiocdev.Line
+  var err error
+
+  fmt.Printf("Initializing line %d...\n", sysGpio)
+  if sysGpio >= 400 {
+    line, err = gpiocdev.RequestLine("gpiochip2", sysGpio - 400, options...)
+  } else {
+    line, err = gpiocdev.RequestLine("gpiochip0", sysGpio, options...)
+  }
+  if err != nil {
+    return nil, fmt.Errorf("error initializing line %v; %v", sysGpio, err)
+  }
+  fmt.Printf("Initialized line %d\n", sysGpio)
+
+  return line, nil
+}
+
+func CloseLine(line *gpiocdev.Line) error {
+  // Add 400 if chip 2
+  chip := line.Chip()
+  lineOffset := line.Offset()
+  if chip == "gpiochip2" {
+    lineOffset += 400
+  }
+
+  fmt.Printf("Closing line %d...\n", lineOffset)
+  err := line.Close()
+  if err != nil {
+		fmt.Printf("Line %d is already closed\n", lineOffset)
+	}
+  fmt.Printf("Closed line %d\n", lineOffset)
+  return nil
+}
+
+func NewGpioChardev(gpioPath string, cfg []config.PinConfig) (*GpioChardev, error) {
+  fmt.Printf("Initializing Chardev Gpio interface\n")
+	if gpioPath != "" {
+		ctrl.sysGpioPath = gpioPath
+	}
+
+  gpios := make(map[int]pin)
+	for _, val := range cfg {
+    newPin := pin{sysNum: val.SysGpio, description: val.Description}
+    gpios[val.ID] = newPin
+	}
+
+	ctrlChardev := GpioChardev{
+		sysGpioPath: gpioPath,
+		gpios:       gpios,
+	}
+
+	return &ctrlChardev, nil
+}
+
+func (ctrl *GpioChardev) SetDirection(id int, direction string) error {
+	pinID := ctrl.GetPinNumByID(id)
+	fmt.Printf("Setting pin %d direction to %v\n", pinID, direction)
+  
+  var line *gpiocdev.Line
```

Comment:
Please take a look how code looks in PR prior submitting. It looks terrible with these indents.